### ISSUE TITLE
Refreshing Project Changes Timestamp in .settings/org.eclipse.buildship.core.prefs

### ISSUE THREAD

**Issue by gchallen** [eclipse/buildship#568]
Being able to import an Eclipse project as a Gradle project requires that the repository contain `.settings/org.eclipse.buildship.core.prefs`. However, ever time the project is refreshed the contents of this file are changed—in particular, a timestamp is altered:

````bash
--- a/.settings/org.eclipse.buildship.core.prefs
+++ b/.settings/org.eclipse.buildship.core.prefs
@@ -1,2 +1,2 @@
-#Sun Sep 03 16:16:19 CDT 2018
+#Tue Sep 12 12:12:10 CDT 2017
 connection.project.dir=
````
This causes Git to repeatedly force you to recommit the file, which is annoying. You can add it to the `.gitignore` file, but then you run the risk of missing it if you use the repository as a skeleton for a new project.

I'm assuming that this timestamp is important somehow. But wouldn't it be possible to have the single setting that this file contains be a default so that this file is not needed at all?

**Comment by donat** [comment]
The problem is that Buildship has two implementations to store preferences. One uses the Eclipse preferences API which seemingly works fine. The other, fallback implementation uses `java.util.Properties` which adds the timestamp. That should be adjusted.

**Comment by donat** [comment]
We could subclass `Properties` and remove the data-writing section from the `store0()` method.

**Comment by gchallen** [comment]
However you can get rid of it is fine with me. This also causes (unnecessary) merge conflicts when merging branches that are being developed independently. So it's annoying on that account as well.

**Comment by donat** [comment]
https://github.com/eclipse/buildship/pull/635 solved this issue too. The fix will be part of 2.2.1 which we plan to release early next week.
